src/inc-init.php

```diff
@@ -476,6 +476,10 @@
 if ($_CONF = $_DB->query('SELECT * FROM ' . TABLE_CONFIG, false, false)) {
     // Must be two-step, since $_CONF can be false and therefore does not have ->fetchAssoc().
     $_CONF = $_CONF->fetchAssoc();
+    if ($_CONF) {
+        // See if the database is read-only at the moment, due to an announcement with the configuration.
+        $_CONF['lovd_read_only'] = (bool) $_DB->query('SELECT COUNT(*) FROM ' . TABLE_ANNOUNCEMENTS . ' WHERE start_date <= NOW() AND end_date >= NOW() AND lovd_read_only = 1')->fetchColumn();
```

Comment:
It's either this, or me updating the `$_CONF` setting every time to reflect the announcement's settings :wink: 
This is, I guess, the lesser of two evils.